### DISCUSSION TITLE
Will the `Store` become an actor?

### DISCUSSION THREAD

**Discussion by krawutzi** [pointfreeco/swift-composable-architecture#444] [General]
Or more generally speaking, how do you think the composable architecture will adopt the new features from the Swift Concurrency Proposal?

**Comment by mbrandonw** [comment]
This is a great question, but unfortunately we haven't had much time to think about it. We will definitely be dedicating a lot of time to it eventually, but not sure when. If anyone has done any explorations we'd love to hear about it.

The only thing I can say is that with concurrent features coming to Swift, I think we'd love if the core TCA had less of a dependency on Combine, and instead Combine/SwiftUI features could be layered on top where necessary. That may mean `Store` becomes an `actor`, reducers become `async` or return `Async*` values. Even better, if this pans out dependencies could break loose from the shackles of Combine/TCA and just vend out simple `async` functions and be pure Swift modules 😍
